frappe/public/js/frappe/form/controls/duration.js

```diff
@@ -0,0 +1,140 @@
+frappe.ui.form.ControlDuration = frappe.ui.form.ControlData.extend({
+	make_input: function() {
+		this._super();
+		this.make_picker();
+	},
+
+	make_picker: function() {
+		this.inputs = [];
+		this.set_duration_options();
+		this.$picker = $(
+			`<div class="duration-picker">
+				<div class="picker-row row"></div>
+			</div>`
+		);
+		this.$wrapper.append(this.$picker);
+		this.build_numeric_input("days", !this.duration_options.showDays);
+		this.build_numeric_input("hours", false);
+		this.build_numeric_input("minutes", false);
+		this.build_numeric_input("seconds", !this.duration_options.showSeconds);
+		this.set_duration_picker();
+		this.$picker.hide();
+		this.bind_events();
+		this.refresh();
+	},
+
+	build_numeric_input: function(label, hidden, max) {
+		let $duration_input = $(`
+			<input class="input-sm duration-input" data-duration="${label}" type="number" min="0" value="0">
+		`);
+
+		let $input = $(`<div class="row duration-row"></div>`).prepend($duration_input);
+
+		if (max) {
+			$duration_input.attr("max", max);
+		}
+
+		this.inputs[label] = $duration_input;
+
+		let $control = $(`
+			<div class="col duration-col">
+				<div class="row duration-row duration-label">${__(label)}</div>
+			</div>`
+		);
+
+		if (hidden) {
+			$control.addClass("hidden");
+		}
+		$control.prepend($input);
+		$control.appendTo(this.$picker.find(".picker-row"));
+	},
+
+	set_duration_options() {
+		this.duration_options = frappe.meta.get_duration_options(this.df);
```

Comment:
Any reason to fetch it from the meta?

P.S. Meta object is not available on website. This will make this control incompatible with WebForms in case we wish to add this in the future

Reply:
moved it to `utils.js`:

![web-form](https://user-images.githubusercontent.com/24353136/82737535-4f803180-9d4f-11ea-95dc-23b159c01ca1.gif)